### PR TITLE
rec: For zones having many NS records take a sample

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

so tak…
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/settings.rst [hunk 1]
```diff
@@ -1224,6 +1224,19 @@ it by the number of NS records found above the
 `max-ns-address-qperq`_ value. The limit wil not be reduced to a
 number lower than 5.
 
+.. _setting-max-nsperq:
```
Comment (rgacogne) [inline pdns/recursordist/docs/settings.rst:1227]: A small nit, I find the `perq` suffix confusing for this setting, as it suggests (to me, at least) that it is the number of NS we are willing to consider per query, while it's actually the number of NS we are considering per zone?

**Review by rgacogne — APPROVED** [review]
Code looks good, I have a small dislike for the setting/internal variable naming but I could live with it.

File: pdns/recursordist/docs/settings.rst (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/settings.rst:1227]: It's wrong indeed. It's per name, I'd say. Once client query might need to resolve several names, each such resolve is subject to the restriction, while max-qperq is _per client query_.

Would the name `max-ns-per-resolve` suit you better? Cannot think of a better name atm.

Comment (rgacogne) [reply pdns/recursordist/docs/settings.rst:1227]: Yes, it would be much better. Thanks!

File: .github/actions/spell-check/expect.txt [hunk 2]
```diff
@@ -1130,6 +1130,7 @@ nsid
 nsis
 nsl
 nsname
+nsperq
```
Comment (rgacogne) [inline .github/actions/spell-check/expect.txt:1133]: This might need an update?

**Pull request merged**